### PR TITLE
Reasons for appropriate adult needed and detainee fit for interview

### PR DESCRIPTION
**Pull request by robmckinnon**
* Change appropriate adult reason input to radio buttons.
* Change `appropriate_adult_reason` from text to string.
* Validate `appropriate_adult_reason` is a valid locales key value.
* Implements story: https://www.pivotaltracker.com/story/show/93690592
* Enable javascript for IE 6.
* Tested show/hide javascript on IE 6/7. Worked ok.

### CODE REVIEW

File: app/helpers/defence_requests_helper.rb [hunk 1]
```diff
@@ -87,4 +87,12 @@ def initial_date_type(initial_date, today, tomorrow)
     end
   end
 
+  def appropriate_adult_reason(defence_request)
+    if defence_request.appropriate_adult?
```
Comment (novotnyjakub) [inline app/helpers/defence_requests_helper.rb]: You don't need to do this check again as the `boolean_with_explanation` helper does it already.

File: spec/models/defence_request_spec.rb [hunk 2]
```diff
@@ -17,6 +17,29 @@
         subject.appropriate_adult = true
       end
       it { expect(subject).to validate_presence_of :appropriate_adult_reason }
+
```
Comment (novotnyjakub) [inline spec/models/defence_request_spec.rb]: Why didn't you use `validate_inclusion_of` Shoulda helper? It should cover all the 3 contexts bellow.

File: app/helpers/defence_requests_helper.rb (hunk 1 above)
Comment (robmckinnon) [inline app/helpers/defence_requests_helper.rb]: I recall getting an error in test runs when this check was not present. This helper code is getting executed before the `boolean_with_explanation` helper code runs.

File: spec/models/defence_request_spec.rb (hunk 2 above)
Comment (robmckinnon) [inline spec/models/defence_request_spec.rb]: Good spot. I'll change this.

File: app/helpers/defence_requests_helper.rb (hunk 1 above)
Comment (novotnyjakub) [inline app/helpers/defence_requests_helper.rb]: Good point, I didn't realise that.

**Comment by novotnyjakub** [comment]
I know we haven't worked on the errors yet, but if you don't select this new radio button, you get this:

`Because the detainee is: can't be blank, is not included in the list`.

Should we leave this until further?

**Comment by robmckinnon** [comment]
At present all errors are appearing in the form `label + error message`, e.g.:
```
Detainee name: can't be blank
Address: can't be blank
...
Is an appropriate adult needed?: can't be blank
Is detainee fit for interview: can't be blank
```
Think it's better to make a new story for implementing custom error messages. Expect we'll end up with a generic solution for doing this.

**Comment by ansonK** [comment]
> I know we haven't worked on the errors yet, but if you don't select this new radio button, you get this:
> 
> Because the detainee is: can't be blank, is not included in the list.
> 
> Should we leave this until further?

Yes. We will go over error messaging as a separate piece of work

**Pull request merged**